docs/website/docs/reference/bindings/c-api.md

```diff
@@ -192,13 +196,172 @@ int main(int argc, char** argv) {
+
+namespace mlir::iree_compiler {
+namespace {
+
+struct TargetDemoOptions {
+  bool demoOption = false;
+  void bindOptions(OptionsBinder &binder) {
+    static llvm::cl::OptionCategory category("Target Demo");
+    binder.opt<bool>(
+        "iree-target-demo-option", demoOption,
+        llvm::cl::cat(category),
+        llvm::cl::desc("Some command-line option"));
+  }
+};
+
+struct TargetDemoSession
+    : public PluginSession<TargetDemoSession, TargetDemoOptions,
+                           PluginActivationPolicy::DefaultActivated> {
+  void populateHALTargetBackends(IREE::HAL::TargetBackendList &targets) {
+    targets.add("demo", [&]() {
+      return std::make_shared<DemoTargetBackend>(options);
+    });
+  }
+};
+
+} // namespace
+} // namespace mlir::iree_compiler
+
+IREE_DEFINE_COMPILER_OPTION_FLAGS(::mlir::iree_compiler::TargetDemoOptions);
+
+extern "C" bool iree_register_compiler_plugin_hal_target_demo(
+    mlir::iree_compiler::PluginRegistrar *registrar) {
+  registrar->registerPlugin<TargetDemoSession>("hal_target_demo");
+  return true;
+}
+```
+
 #### Samples
 
 | Project | Source | Description |
-| ------- |------- | ----------- |
-[iree-org/iree-template-compiler-cmake](https://github.com/iree-org/iree-template-compiler-cmake/) | [`hello_compiler.c`](https://github.com/iree-org/iree-template-compiler-cmake/blob/main/hello_compiler/hello_compiler.c) | Compiler application template
-[openxla/openxla-pjrt-plugin](https://github.com/openxla/openxla-pjrt-plugin/) | [`iree_compiler.cc`](https://github.com/openxla/openxla-pjrt-plugin/blob/main/iree/integrations/pjrt/common/iree_compiler.cc) | JIT for TensorFlow + JAX to IREE
-[openxla/iree](https://github.com/openxla/iree/) | [`samples/compiler_plugins/`](https://github.com/openxla/iree/tree/main/samples/compiler_plugins) | In-tree demos of compiler plugins
+| ------- | ------ | ----------- |
+[iree-org/iree-template-compiler-cmake](https://github.com/iree-org/iree-template-compiler-cmake/) |[`hello_compiler.c`](https://github.com/iree-org/iree-template-compiler-cmake/blob/main/hello_compiler/hello_compiler.c) | Compiler application template
+[openxla/iree](https://github.com/openxla/iree/) | [`integrations/pjrt/.../iree_compiler.cc`](https://github.com/openxla/iree/blob/main/integrations/pjrt/src/iree_pjrt/common/iree_compiler.cc) | JIT for TensorFlow + JAX to IREE
+[openxla/iree](https://github.com/openxla/iree/) | [`compiler/plugins`](https://github.com/openxla/iree/tree/main/compiler/plugins) | In-tree supported compiler plugins
+[openxla/iree](https://github.com/openxla/iree/) | [`samples/compiler_plugins/`](https://github.com/openxla/iree/tree/main/samples/compiler_plugins) | In-tree sample compiler plugins
+[nod-ai/iree-amd-aie](https://github.com/nod-ai/iree-amd-aie/) | [`plugins/.../iree-amd-aie`](https://github.com/nod-ai/iree-amd-aie/tree/main/compiler/plugins/target/AMD-AIE/iree-amd-aie) | Early-phase plugins for interfacing with AMD AIE accelerators
```

Comment:
Calling this out in case you miss it scanning - this adds a link to iree-amd-aie as a sample use of the compiler APIs
![image](https://github.com/openxla/iree/assets/4010439/b69985b9-3a6a-44c6-b838-fa0fe8d57437)